### ISSUE TITLE
Unit test failure on Moodle 3.5: local_o365_webservices_onenoteassignment_testcase::test_assignment_read

### ISSUE THREAD

**Issue by jwalits** [microsoft/o365-moodle#1218]
Unit test failures posted below. 

First one, I believe it is getting caused by these 2 lines:

https://github.com/microsoft/moodle-local_o365/blob/master/tests/webservices_onenoteassignment_test.php#L188
https://github.com/microsoft/moodle-local_o365/blob/master/tests/webservices_onenoteassignment_test.php#L196

The 2nd failure is getting caused by these lines:

https://github.com/microsoft/moodle-local_o365/blob/master/tests/webservices_utils_test.php#L103
https://github.com/microsoft/moodle-local_o365/blob/master/tests/webservices_utils_test.php#L123

Its passing an object when a string is expected.

Added a fix in the below commit. Please review and let me know any feedback

```
root@230577df9717:/siteroot# vendor/bin/phpunit "local_o365_webservices_onenoteassignment_testcase" local/o365/tests/webservices_onenoteassignment_test.php
Moodle 3.5.10+ (Build: 20200214), 59377c37977894892ad5b5daca7e8164c921fad6
Php: 7.2.24.0.0.18.04.2, pgsql: 11.5, OS: Linux 5.3.0-28-generic x86_64
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

....E.......                                                      12 / 12 (100%)

Time: 3.43 seconds, Memory: 94.75MB

There was 1 error:

1) local_o365_webservices_onenoteassignment_testcase::test_assignment_read
PHPUnit\Framework\Exception: Argument #1 (No Value) of PHPUnit\Framework\TestCase::expectExceptionMessage() must be a string

/siteroot/local/o365/tests/webservices_onenoteassignment_test.php:310
/siteroot/lib/phpunit/classes/advanced_testcase.php:80
```
```
There was 1 error:

1) local_o365_webservices_utils_testcase::test_get_assignment_info with data set "Course not found (no course)" (false, array(array('10', 'Test!', 'test')), array('dml_missing_record_exception', lang_string Object (...)), 1, 10, array())
PHPUnit\Framework\Exception: Argument #1 (No Value) of PHPUnit\Framework\TestCase::expectExceptionMessage() must be a string

/siteroot/local/o365/tests/webservices_utils_test.php:281
/siteroot/lib/phpunit/classes/advanced_testcase.php:80
```

**Comment by avegys** [comment]
Hi,

Thank you for your reported issue. The fix is on the way to be released soon.

Best regards,
Andrius

**Comment by avegys** [comment]
Hi jwalits,

Thank you for your efforts helping to improve Office365 plugins. Your code changes that fixes test has been included into latest release for Moodle 3.7 and Moodle 3.8.

Bets regards,
Andrius
